compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1752,9 +1752,9 @@ private void populateIntersectionTypeReferencedFunctions(DataInputStream inputSt
             String attachedFuncName = getStringCPEntryValue(inputStream);
             String attachedFuncOrigName = getStringCPEntryValue(inputStream);
             var attachedFuncFlags = inputStream.readLong();
+            BInvokableType attachedFuncType = (BInvokableType) readTypeFromCp();
```

Comment:
What's this change about? We anyway use the read function only within the if check

Reply:
Do we need to skip some part of the byte stream for some cases?

Reply:
In BIRTypeWriter we didn't use such kind of logic(check whether symbol contains `interface` or `attached` flags).
https://github.com/ballerina-platform/ballerina-lang/blob/05522577aeae55e6974938886ee10842c1f4ade1/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java#L446